### DISCUSSION TITLE
New to Beaker and not getting it 100%

### DISCUSSION THREAD

**Discussion by ghost** [beakerbrowser/beaker#1808] [General] [answered]
Hello, I've been tinkering (that's all my skills really allow me to do lol) and LOVE the concept of Beaker. What a great idea and I'm looking forward to seeing where this goes. That being said I seem to be missing some basic information. 

I've looked through the documentation (https://docs.beakerbrowser.com/) and don't see any information about who I am able to share my hyperdrive with. It seems I just need to "copy the drive URL" and am then able to share it from there. But when I try to access the hyperdrive from Brave or Edge (pretending to be the recipient of my hyperdrive link) I am unable to. Those browsers just deal with the URL (hyper://c22bfccccca2813...etc/) as a search string to which it finds no results. This leads me to believe a number of things; I just don't grasp the idea and have missed a basic setting or permission or concept, and secondly that the user I wish to share with MUST have Beaker. I'm 100% on the first part, I'm just a general computer lover who likes to tinker with various software and have no real coding knowledge or ability. So it's easy to believe I've missed something. On the second point I have done some pretty decent (though not exhaustive) searches here in Discussions and Issues as well as the published documentation on the beaker site and here and don't see it say anywhere that the person I am sharing with HAS to have Beaker. Nor is it even generally implied except that everyone here talking about sharing hyperdrives via Beaker....IS using Beaker. 

I'm not sure how to proceed to share my hyperdrive based site, is there a way to share it to a friend who doesn't have Beaker? (I see in a post here where it does seem other people do, or have a reasonable expectation to, using a link in an html file). Or do I need to have my friend install beaker to be able to access the files I'd like to share with them? While I wouldn't mind encouraging them to do that, it seems to not meet the expectation I had after reading what Beaker was capable of and it's ease of use. 

Aside from the above, in creating my first hyperdrive that is shareable (my personal drive is not as per Beaker), I found that loading files into it took a VERY long time. There is about 5.5Gb of data in about 15 files that took almost 5 hours to copy/sync to the hyperdrive. At first I tried the option to create a new drive from a folder as that seemed the most efficient method for my needs (I had a folder with files I wanted to share, all I needed to do was create an index.html file... which I did in Sublime Text) but after trying to sync that folder I canceled after 3 hours and tried an alternate method - Creating  new drive from within Beaker then dragging files in from Windows File Explorer. This method was preferable in that it gave a progress bar and I was able to see that something WAS happening, albeit achingly slowly (it may have been happening in the first tried method but it was impossible to tell given there was no progress bar). I'm not sure that anyone else has had this specific issue with hyperdrive creation but I do see some sort of similar issues here. 

Thirdly overall , one thing I'd like to be able to do is pick the drive location for my hyperdrives, on an external HDD for example. I don't see this option anywhere. Again I may be missing something (a setting or a security concept I haven't given full consideration).

Thanks for any help.

**Comment by pfrazee** [comment, marked as answer]
> On the second point I have done some pretty decent (though not exhaustive) searches here in Discussions and Issues as well as the published documentation on the beaker site and here and don't see it say anywhere that the person I am sharing with HAS to have Beaker. Nor is it even generally implied except that everyone here talking about sharing hyperdrives via Beaker....IS using Beaker.

Yeah it's not so much that `hyper://` only works on Beaker. It's just that no mainstream browsers currently support `hyper://`. It's a new technology and it's going to take time to gain adoption.

> I'm not sure how to proceed to share my hyperdrive based site, is there a way to share it to a friend who doesn't have Beaker? (I see in a post here where it does seem other people do, or have a reasonable expectation to, using a link in an html file). Or do I need to have my friend install beaker to be able to access the files I'd like to share with them? While I wouldn't mind encouraging them to do that, it seems to not meet the expectation I had after reading what Beaker was capable of and it's ease of use.

It's good to know that we could help make that clearer to people, but yeah currently you're going to need Beaker to access the files.

> Aside from the above, in creating my first hyperdrive that is shareable (my personal drive is not as per Beaker), I found that loading files into it took a VERY long time. There is about 5.5Gb of data in about 15 files that took almost 5 hours to copy/sync to the hyperdrive.

Yikes. We'll have to look into that. 5.5Gb is plenty of data but that's way longer than makes any sense.

> Thirdly overall , one thing I'd like to be able to do is pick the drive location for my hyperdrives, on an external HDD for example. I don't see this option anywhere. Again I may be missing something (a setting or a security concept I haven't given full consideration).

We don't support that yet, there's an internal "database" where all the hyper data goes, and that can't currently be configured.

Hope this helps!

**Comment by ghost** [comment]
Thank you so much for your answers. It helps greatly. I will encourage my friend to use Beaker for our sharing interactions. They're less adventurous than I am but it would make life easy for them and I to share stuff.

One thing I have noticed now is that my .hyperdrive folder (C:\Users\myusername\ .hyperdrive) is now almost 10Gb. My only hyperdrive being the mentioned 5.5Gb. I get the feeling that there is leftover data from the failed hyperdrive creation mentioned as the first attempt in my original posting. Is there a way to clear this? Like deleting the .hyperdrive folder and starting over? Or will that delete my personal hyperdrive as well?

Anyways, thanks again for the reply I know you're busy working on this great project. Your time, focused on my questions, is appreciated.

**Reply by pfrazee** [reply]
You're 100% right about what's going on in the `.hyperdrive` folder and the issue of deleting it. Again, something else we gotta do -- have a way to clear out unused data. If you do decide to clear it out, you'll need to clear all beaker data, which you can do by going to `beaker://settings/?view=info` and finding your "user data" directory's path, and delete that.

Thanks for the kind words, especially in light of the fairly nonsense problems you're running into.

**Reply by daya** [reply]
> finding your "user data" directory's path

so i created a first wiki called `daya-wiki` where should I expect to find it
